ee/clickhouse/models/test/test_filters.py

```diff
@@ -740,3 +740,66 @@ def test_person_cohort_properties(self):
         result = sync_execute(query, {"team_id": self.team.pk, **prop_clause_params})[0][0]
 
         self.assertEqual(result, person2_distinct_id)
+
+    def test_simplify_nested(self):
+        filter = Filter(
+            data={
+                "properties": {
+                    "type": "OR",
+                    "values": [
+                        {
+                            "type": "OR",
+                            "values": [
+                                {
+                                    "type": "AND",
+                                    "values": [
+                                        {"type": "person", "key": "email", "operator": "icontains", "value": ".com",}
+                                    ],
+                                }
+                            ],
+                        },
+                        {
+                            "type": "AND",
+                            "values": [
+                                {"type": "person", "key": "email", "operator": "icontains", "value": "arg2",},
+                                {"type": "person", "key": "email", "operator": "icontains", "value": "arg3",},
+                            ],
+                        },
+                    ],
+                }
+            }
+        )
+
+        # Can't remove the single prop groups if the parent group has multiple. The second list of conditions becomes property groups
+        # because of simplify now will return prop groups by default to ensure type consistency
+        self.assertEqual(
+            filter.simplify(self.team).properties_to_dict(),
+            {
+                "properties": {
+                    "type": "OR",
+                    "values": [
+                        {
+                            "type": "OR",
+                            "values": [{"type": "person", "key": "email", "operator": "icontains", "value": ".com",}],
+                        },
+                        {
+                            "type": "AND",
+                            "values": [
+                                {
+                                    "type": "AND",
+                                    "values": [
+                                        {"type": "person", "key": "email", "operator": "icontains", "value": "arg2",}
+                                    ],
+                                },
+                                {
+                                    "type": "AND",
+                                    "values": [
+                                        {"type": "person", "key": "email", "operator": "icontains", "value": "arg3",}
+                                    ],
+                                },
+                            ],
+                        },
+                    ],
+                }
+            },
+        )
```

Comment:
@neilkakkar it behaves like this. So it doesn't get to simplify everything perfectly we still get excess groups if the parent group has multiple. In this case, you'll notice that the properties actually became prop groups (due to the changes in this PR) and could not get simplified because the parent group has multiple values. But it still simplified the first group that had a single value

Reply:
makes sense I think